pkg/didcomm/envelope/api.go

```diff
@@ -0,0 +1,51 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package envelope
+
+// Package envelope manages the handling of DIDComm raw messages in JWE compliant envelopes.
+// The aim of this package is build these envelopes and parse them. They are mainly used as
+// wire-level wrappers of 'payloads' used in DID Exchange flows.
+
+// PackagerCreator method to create new outbound dispatcher service
+type PackagerCreator func(prov Provider) (Packager, error)
+
+// Packager provide methods to pack and unpack msg
+type Packager interface {
+	// PackMessage Pack a message for one or more recipients.
+	//
+	// Args:
+	//
+	// envelope: The message to pack
+	//
+	// Returns:
+	//
+	// []byte: The packed message
+	//
+	// error: error
+	PackMessage(envelope *Envelope) ([]byte, error)
+
+	// UnpackMessage Unpack a message.
+	//
+	// Args:
+	//
+	// encMessage: The encrypted message
+	//
+	// Returns:
+	//
+	// envelope: unpack message
+	//
+	// error: error
+	UnpackMessage(encMessage []byte) (*Envelope, error)
+}
+
+// Envelope contain msg, FromVerKey and ToVerKeys
+type Envelope struct {
+	Message    []byte
+	FromVerKey string
```

Comment:
"From" and "To" are relative to the direction the envelope is going (inbound vs outbound) and might get confusing. I suggest we keep it simple: 

* FromVerKey -> MyVerKey
* ToVerKeys -> TheirVerKeys

Reply:
I don't see how MyVerKey is clearer than FromVerKey? 

This will be more confusing from the agent's perspective since `MyVerKey` for a receiving agent is going to be different from the sending agent's while `FromVerKey` is clear to be the sender's agent key, no matter who is using the Envelope (`From` will always be source agent).

Again, this code was moved from `wallet` into `envelope`, it's not new code.

Reply:
`Envelope` is exported for reasons I don't understand.

If all the complexity of handling `Envelope` structs is contained within the packing/unpacking functionality then it's okay. But if `Envelope` starts percolating into the application layer where code would have to first determine their role (sender vs receiver) before picking the right attribute ("from" vs "to") - that would be an unnecessary burden imposed on the application layer.